src/index.js

```diff
@@ -0,0 +1,96 @@
+const addTaskButton = document.getElementById('add-task-btn');
+const newTaskInput = document.getElementById('new-task-input');
+const todoList = document.getElementById('todo-list');
+const clearCompletedButton = document.getElementById('clear-completed-btn');
+
+function deleteTask(todoItem) { todoItem.remove(); }
+
+function editTask(todoItem) {
+  const taskElement = todoItem.querySelector('.task');
+  const taskText = taskElement.textContent;
+  taskElement.innerHTML = '';
+  const editInput = document.createElement('input');
+  editInput.type = 'text';
+  editInput.value = taskText;
+  editInput.classList.add('edit-input');
+  taskElement.appendChild(editInput);
+  editInput.focus();
+  editInput.addEventListener('keyup', (event) => {
+    if (event.key === 'Enter') {
+      const newText = editInput.value.trim();
+      if (newText !== '') {
+        taskElement.textContent = newText;
+        todoItem.style.backgroundColor = '';
+        todoItem.querySelector('.more-btn').style.display = 'inline-block';
+        todoItem.querySelector('.delete-btn').style.display = 'none';
+      }
+    }
+  });
```

Comment:
- [ ] As per project requirements, you were supposed to create an index.js file and set an array of some simple to do tasks (array of objects). Each task object should contain three keys.
![image](https://github.com/Hsuwaddyhtat22/To-Do-List-mini-PJ/assets/62849872/f0a56bc0-02fa-4273-b726-9b479af6a076)

There is no tasks array explicitly defined. However, the code creates a todo list by dynamically creating an HTML list and adding items to it. Each item in the list is represented by an HTML li element, which contains a checkbox, a task description, and two buttons (one to show more options and another to delete the task). Kindly add the array as per project requirements.
**Example**
```javascript
const tasks = [
  {
    description: 'An example description in string',
    completed: false,
    index: 10,
  },
];
```